### ISSUE TITLE
[Feature] Add secret storage in project 

### ISSUE THREAD

**Issue by ketchoop** [php-censor/php-censor#14]
It's just a killer feature that PHPCI hasn't (but a lot of CIs has).

## How it works:

* You add secrets in project settings
* They had been injected in system environment variables before project builds
* You can take them from env

## Use cases:

Add keys, passwords to deploy or something like these.

## Why it's better than inject them by env plugin:

Because of security. It's not good idea to store secrets in repo(it can be public).

## What's needed to change

* Database schema. Add table for secrets and add connections between project and secrets.
* Project view. Add fields for it.
* Form builder. Add multiple inputs that can be fetched as array.
* Build logic. Add fetching and injecting(exporting).

What do you think?

P.S. Maybe I'll try to implement this if I have free time.

**Comment by corpsee** [comment]
It is very useful feature! It would be great if you have implemented it.

P.S.: Now for this behavior you may use project config from project settings. It will replace project config from repository.

**Comment by ket4yii** [comment]
According to #41 it can be closed.

**Comment by corpsee** [comment]
Hmm... Is #41 the same?? I was going to finish the secret storage by myself (and use your branch and code).

**Comment by ket4yii** [comment]
Hmmm... Maybe #41 it will be not the same, if you use some encryption for example(like hashicorp vault). But that issue inject variables and allows you to use it in project, that looks the same as my idea. Anyway, maybe I'm not right.

**Comment by ket4yii** [comment]
Or just use annotation(or yaml tags) for #41. Like 
```
my_super_secret_var: !secret TOP SECRET 
```

and after that it'll be:

```
my_super_secret_var: *******
```

**Comment by ket4yii** [comment]
Sorry, it's my misunderstanding of #41. It's not about secrets. But my last comment will be good point about secrets(I like UI such as it;s in Travis, but with PHPCI framework doing it is torture)

**Comment by corpsee** [comment]
:smile:

**Comment by bochkovprivate** [comment]
+ 1 like.
It would be great if I could reference a secret from main yaml and that reference would be substituted by the value corresponding to the current project / environment.

**Comment by corpsee** [comment]
I close the issue for now because the most important things are implemented. See last comment in #87.

**Comment by corpsee** [comment]
Implemented in `master` branch. Will be released with version `2.1.0`.
